src/actions/index.js

```diff
@@ -46,25 +45,28 @@ const ALERT_LEVEL_ERROR = 'error'
 export const downloadFileMissing = () => ({ type: DOWNLOAD_FILE_E_MISSING, alert: { message: 'error.download_file.missing', level: ALERT_LEVEL_ERROR } })
 export const downloadFileOffline = () => ({ type: DOWNLOAD_FILE_E_OFFLINE, alert: { message: 'error.download_file.offline', level: ALERT_LEVEL_ERROR } })
 
-export const openFolder = (folderId, context = FILES_CONTEXT) => {
+export const locationChange = (virtualRoot, folderId) => {
   return async dispatch => {
-    if (!folderId) {
-      folderId = context === TRASH_CONTEXT
-        ? TRASH_DIR_ID
-        : ROOT_DIR_ID
-    }
-    dispatch({ type: OPEN_FOLDER, folderId, context })
+    dispatch({ type: LOCATION_CHANGE, virtualRoot, folderId })
+    return dispatch(openFolder(virtualRoot, folderId))
+  }
+}
+
+export const openFolder = (virtualRoot, folderId) => {
+  return async dispatch => {
+    if (!folderId) folderId = virtualRoot
+    dispatch({ type: OPEN_FOLDER, virtualRoot, folderId })
```

Comment:
`virtualRoot` is the new `context`... The problem is that we need to keep the view's "type" (normal or trash) or "root" (/files or /trash) in the store... I can't find a better way to implement the specified nav behavior.